pkg/api/store/cluster/cluster_store.go

```diff
@@ -17,3 +25,66 @@ func (r *Store) ByID(apiContext *types.APIContext, schema *types.Schema, id stri
 	}
 	return r.Store.ByID(apiContext, schema, id)
 }
+
+func (r *Store) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
+
+	name, ok := data["name"].(string)
+	if !ok {
+		return nil, httperror.NewFieldAPIError(httperror.MissingRequired, "Cluster name", "")
+	}
+
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	if err := canUseClusterName(apiContext, name); err != nil {
+		return nil, err
+	}
+
+	return r.Store.Create(apiContext, schema, data)
+}
+
+func (r *Store) Update(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, id string) (map[string]interface{}, error) {
+
+	existingCluster, err := r.ByID(apiContext, schema, id)
+
+	if err != nil {
+		return nil, err
+	}
+
+	clusterName, ok := existingCluster["name"].(string)
+	if !ok {
+		clusterName = ""
+	}
+
+	updatedName, ok := data["name"].(string)
```

Comment:
we should use convert library, otherwise there are chances of nil pointer. I believe it's convert.toString()

Reply:
Will check the library!